x-pack/plugins/fleet/server/services/epm/archive/storage.ts

```diff
@@ -147,89 +139,40 @@ export async function getAsset(opts: {
-  }
+export async function getAssetsFromReferences(opts: {
+  savedObjectsClient: SavedObjectsClientContract;
+  references: PackageAssetReference[];
+}): Promise<PackageAsset[]> {
+  const { savedObjectsClient, references } = opts;
+  const bulkRes = await savedObjectsClient.bulkGet<PackageAsset>(references);
+  return bulkRes.saved_objects.map((so) => so.attributes);
+}
+
+export function packageAssetToArchiveEntry(asset: PackageAsset): ArchiveEntry {
+  const { asset_path: path, data_utf8: utf8, data_base64: base64 } = asset;
+  const buffer = utf8 ? Buffer.from(utf8, 'utf8') : Buffer.from(base64, 'base64');
 
-  let dataStreamPaths: string[] = [];
-  const dataStreams: RegistryDataStream[] = [];
-  paths
-    .filter((path) => path.startsWith(`${pkgKey}/data_stream/`))
-    .forEach((path) => {
-      const parts = path.split('/');
-      if (parts.length > 2 && parts[2]) dataStreamPaths.push(parts[2]);
-    });
-
-  dataStreamPaths = uniq(dataStreamPaths);
-
-  await Promise.all(
-    dataStreamPaths.map(async (dataStreamPath) => {
-      const dataStreamManifestPath = `${pkgKey}/data_stream/${dataStreamPath}/manifest.yml`;
-      const soResDataStreamManifest = await savedObjectsClient.find<PackageAsset>({
-        type: ASSETS_SAVED_OBJECT_TYPE,
-        perPage: 1,
-        page: 1,
-        filter: `${ASSETS_SAVED_OBJECT_TYPE}.attributes.package_name:${pkgName} AND ${ASSETS_SAVED_OBJECT_TYPE}.attributes.package_version:${pkgVersion} AND ${ASSETS_SAVED_OBJECT_TYPE}.attributes.asset_path:${dataStreamManifestPath}`,
-      });
-      const dataStreamManifest = yaml.load(
-        soResDataStreamManifest.saved_objects[0].attributes.data_utf8
-      );
-
-      const {
-        title: dataStreamTitle,
-        release,
-        ingest_pipeline: ingestPipeline,
-        type,
-        dataset,
-      } = dataStreamManifest;
-      const streams = parseAndVerifyStreams(dataStreamManifest, dataStreamPath);
-
-      dataStreams.push({
-        dataset: dataset || `${pkgName}.${dataStreamPath}`,
-        title: dataStreamTitle,
-        release,
-        package: pkgName,
-        ingest_pipeline: ingestPipeline || 'default',
-        path: dataStreamPath,
-        type,
-        streams,
-      });
-    })
-  );
-  packageInfo.policy_templates = parseAndVerifyPolicyTemplates(packageInfo);
-  packageInfo.data_streams = dataStreams;
   return {
-    paths,
-    packageInfo,
+    path,
+    buffer,
   };
-};
+}
+
+export async function getEsPackage(opts: {
+  references: PackageAssetReference[];
+  savedObjectsClient: SavedObjectsClientContract;
+}) {
+  const { references, savedObjectsClient } = opts;
+  const assets = await getAssetsFromReferences({ references, savedObjectsClient });
+  const entries: ArchiveEntry[] = assets.map(packageAssetToArchiveEntry);
+  const packageResponse = parseAndVerifyEntries(entries);
+
+  entries.forEach(({ path, buffer }) => {
+    if (path && buffer) {
+      setArchiveEntry(path, buffer);
```

Comment:
Added this to mirror the approach we take during package installation of adding all items to the cache. We'll want to revisit this as part of a review of how we cache but it's consistent